File: MyDraw/app/src/main/java/com/example/mydraw/MainActivity.java
package com.example.mydraw;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    private SingleTouchView drawView;
    private ImageButton currPaing;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        drawView = (SingleTouchView) findViewById(R.id.drawing);
        LinearLayout paintLayout = (LinearLayout) findViewById(R.id.paingColor); //색깔펜이 들어있는 리니얼레이아웃을 가지고 온다
        currPaing = (ImageButton) paintLayout.getChildAt(0); //그리고 그 리니얼레이아웃의 index 0번을 가져온다.
    }

    public void colorpan(View view){
        if(view!=currPaing){ //currPaing이 지금 view가 아니라면
            String color = view.getTag().toString(); //그 색깔펜의 색을 받아오고
            drawView.setColor(color); // 그 색을 셋팅해준다
            currPaing = (ImageButton) view; //그리고 currPaing에 그 화면을 보여준다
        }
    }
}

File: KTBook801/app/src/main/java/com/example/ktbook801/MainActivity.kt
package com.example.ktbook801

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.Constraints.TAG
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


class MainActivity : AppCompatActivity() {
    // : 는 extends를 의미. 클래스에 ()을 붙이는 것도 자바랑 다름!
    // fun은 function
    // ?가 있다면 null도 가능하다.
    // ?이 없다면 null-fafe(null이 불가능)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //레트로핏 구현체 생성
        val retrofit = Retrofit.Builder()
            .baseUrl("https://book.interpark.com")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        val bookService = retrofit.create(BookService::class.java)

        bookService.getBestSellerBooks("인터파크도서키를 여기에 넣으면 된다")
            .enqueue(object: Callback<BestSellerDTO>{
                override fun onResponse(
                    call: Call<BestSellerDTO>,
                    response: Response<BestSellerDTO>
                ){
                    if(response.isSuccessful.not()){
                        Log.e(TAG, "Noo!!!!!!seuccess!!!!!!!!!")
                        return;
                    }
                    response.body()?.let{
                        Log.d(TAG, "body에서 꺼내오기"+it.toString())
                        it.books.forEach({ book ->
                            Log.d(TAG,book.toString())
                        })
                    }
                }

                //실패 처리를 선언해주지 않으면 에러가 생긴다
                override fun onFailure(call: Call<BestSellerDTO>, t: Throwable) {
                    //실패처리
                    Log.e(TAG, t.toString())
                }

            })
    }
}